src/Core/maps/src/Maps/Handlers/Map/MapHandler.iOS.cs

```diff
@@ -0,0 +1,170 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using CoreLocation;
+using MapKit;
+using Microsoft.Maui.Dispatching;
+using ObjCRuntime;
+using UIKit;
+using Microsoft.Maui.Handlers;
+using System.Drawing;
+using Microsoft.Maui.Maps.Platform;
+using Microsoft.Maui.Platform;
+using System.Runtime.CompilerServices;
+
+namespace Microsoft.Maui.Maps.Handlers
+{
+	
+	public partial class MapHandler : ViewHandler<IMap, MKMapView>
+	{
+		CLLocationManager? _locationManager;
+		UITapGestureRecognizer? _mapClickedGestureRecognizer;
+
+		protected override MKMapView CreatePlatformView()
+		{
+			return MapPool.Get() ?? new MauiMKMapView(this);
+		}
+
+		protected override void ConnectHandler(MKMapView platformView)
+		{
+			base.ConnectHandler(platformView);
+			_locationManager = new CLLocationManager();
+
+			PlatformView.AddGestureRecognizer(_mapClickedGestureRecognizer = new UITapGestureRecognizer(OnMapClicked));
+
+			if (platformView is MauiMKMapView customMKMapView)
+				customMKMapView.LayoutSubviewsFired += CustomMKMapViewLayoutSubviewsFired;
+
+			PlatformView.RegionChanged += MkMapViewOnRegionChanged;
+			//PlatformView.OverlayRenderer = GetViewForOverlay;
+
+			var mapsPinsItemsSource = (ObservableCollection<IMapPin>)VirtualView.Pins;
+			mapsPinsItemsSource.CollectionChanged += OnPinCollectionChanged;
```

Comment:
These changes need to propagate from the virtual view not via INCC. 

Two approaches here
- If you look at the picker control you'll see at the controls level we call `Handler.UpdateValue(ItemSource)` if the picker values change
- If you want more granular control over adding/removing items than I'd look at the way Layout or the FlyoutMenu handlers are done where they have specific commands for "add/remove/insert/remove'

Reply:
Awesome thanks for the tip, i wasn't sure also about this one